### DISCUSSION TITLE
How to submit an Offer to a contract using PurseInvitationSpec

### DISCUSSION THREAD

**Discussion by dillu24** [Agoric/agoric-sdk#8841] [Q&A]
Let's say that I have a basic hello world contract which invites specific users to print arguments. I managed to deposit a `printArgs` invitation to my wallet and now I want to submit an offer to my contract programatically or using a shell script, passing custom `offerArgs` inside my offer. Am I on the right track? I am particularly not sure what `instance` should be.

```
let templateOffer = {
    invitationSpec: {
      source: "purse",
      description: "printArgs",
      instance: <not-sure-what-this-should-be>
    },
    offerArgs: "Hello World",
    id: Number(Date.now()),
    proposal: {},
};
```

**Comment by dckc** [comment]
Do us a favor? take a look at [Specifying Offers](https://docs.agoric.com/guides/getting-started/contract-rpc.html#specifying-offers), especially [Marshalling Amounts and Instances](https://docs.agoric.com/guides/getting-started/contract-rpc.html#marshalling-amounts-and-instances) and give us some feedback?

We aim to do some work on it to update it w.r.t. ui-kit. But meanwhile, if you like, you can take a look at what we have.

**Comment by dckc** [comment]
a detail:

> ```js
>     offerArgs: "Hello World",
> ```

I think `offerArgs` has to be an `{...}` object.

_IOU citation_

**Comment by dillu24** [comment]
Will have a look at the resources, thanks for that.

FYI I had managed to make this work as follows:

```
let templateOffer = {
    invitationSpec: {
      source: "purse",
      description: "printArgs",
      instance: fromBoard.convertSlotToVal('board063107', "Alleged: InstanceHandle"),
    },
    offerArgs: harden({test: "Hello World"}),
    id: Number(Date.now()),
    proposal: {},
};
```

**Comment by dckc** [comment]
> FYI I had managed to make this work as follows:
> 
> ```
>       instance: fromBoard.convertSlotToVal('board063107', "Alleged: InstanceHandle"),
> ```

bingo! well done.
